lib/apivore/swagger.rb

```diff
@@ -34,11 +34,16 @@ def each_response(&block)
             raise "No responses found in swagger for path '#{path}', " \
               "method #{verb}: #{method_data.inspect}"
           end
+
           method_data.responses.each do |response_code, response_data|
-            schema_location = nil
-            if response_data.schema
-              schema_location = Fragment.new ['#', 'paths', path, verb, 'responses', response_code, 'schema']
-            end
+            schema_location =
+              case
+              when response_data.schema
+                Fragment.new(['#', 'paths', path, verb, 'responses', response_code, 'schema'])
+              when (ref = response_data['$ref'])
+                Fragment.new(ref.split('/'))
```

Comment:
what's the result here? seems you are missing `'schema'`

```ruby
Fragment.new(ref.split('/').push('schema'))
```

Reply:
@lucasrenan I'm not sure why would we missing a `'schema'`. `ref` holds the value of the path to the schema definition. Eg.: in the added test it's `'#/definitions/service'`. And that's exactly the path to the schema definition at the end of file:  https://github.com/westfieldlabs/apivore/pull/89/files#diff-65c886d34e6323742cba1cd52731ecc5R129